kp.go

```diff
@@ -80,6 +82,32 @@ type ClientConfig struct {
 	Timeout       float64 // KP request timeout in seconds.
 }
 
+func queryParamReqOpt(key string, val string) reqOpt {
+	return func(r *http.Request) {
+		q := r.URL.Query()
+		q.Set(key, val)
+		r.URL.RawQuery = q.Encode()
+	}
+}
+
+type reqOpt func(r *http.Request)
+
+func WithForce(force bool) reqOpt {
+	return queryParamReqOpt("force", strconv.FormatBool(force))
+}
+
+func WithLimit(limit int) reqOpt {
+	return queryParamReqOpt("limit", strconv.Itoa(limit))
+}
+
+func WithOffset(offset int) reqOpt {
+	return queryParamReqOpt("offset", strconv.Itoa(offset))
+}
+
+func WithTotalCount(totalCount bool) reqOpt {
+	return queryParamReqOpt("totalCount", strconv.FormatBool(totalCount))
+}
```

Comment:
@Cameron-Kurotori here is my thought. 

- Since the parameters such as limit, offset, totalCount are more for controlling the datasets for any Key Protect API that supports pagination, they can be defined at a common place with the common parameter names similar to how it is documented [here](https://cloud.ibm.com/apidocs/key-protect#pagination) in the KP API docs . 
- The parameters which has API specific context (ex: `state` and `extractable` for listKeys) can be defined with more contextual names (like `WithKeyState`, `WithExtractableKeys`) but they can still be in common place (`kp.go`) so that all reqOpts are grouped at one place similar to how we grouped the query params in api-docs yaml file.

Reply:
from my perspective, as a user of an SDK, I would expect to find the parameters I can use with a specific API to be defined at the same place the API is defined.

Perhaps what we can do is have the base definition all colocated in kp.go but near the API's create variables specific for each API that references them.


```
func WithOffset(offset int) reqOpt {
	return queryParamReqOpt("offset", strconv.Itoa(offset))
}
```


```
// reqOpt generators for List Keys
var (
    WithListKeysLimit = WithLimit
    WithListKeysOffset = WithOffset
    WithListKeysTotalCount = WithTotalCount
    WithListKeysStateFilter = ...
    WithListKeysExtractableFilter = ...
)

func ListKeys(opts ...reqOpt)  ([]Key, error) {...}
```